### ISSUE TITLE
AttributeError: 'NoneType' object has no attribute 'numel'

### ISSUE THREAD

**Issue by simomounir** [scverse/scvi-tools#2243]
Hi,

I have been running into this bug for a while. All I did was to download new data and try to get a latent space projection with GIMVI model, but it fails as soon as I want to start training with **model.train()**.

I followed the same exact steps almost as the jupyter notebook under this link below:

https://docs.scvi-tools.org/en/0.19.0/tutorials/notebooks/gimvi_tutorial.html

I only replaced the cortext() and smfish() data with my own test data (an sc-rnaseq reference dataset and a spatial smfish dataset generated by collaborators. The same preprocessing methods were performed. 

```python
# Your code here
```

```pytb
**AttributeError: 'NoneType' object has no attribute 'numel'**
```

#### Versions:

> '1.0.2'

**Comment by martinkim0** [comment]
Hi, could you paste the full traceback of the error here? Thanks!

**Comment by simomounir** [comment]
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[66], line 3
      1 #model = GIMVI(seq_data_0, spatial_data_0)
      2 # train for 200 epochs
----> 3 model.train(200)

File [c:\Users\resolve\.conda\envs\scvi-env\lib\site-packages\scvi\external\gimvi\_model.py:250](file:///C:/Users/resolve/.conda/envs/scvi-env/lib/site-packages/scvi/external/gimvi/_model.py:250), in GIMVI.train(self, max_epochs, use_gpu, accelerator, devices, kappa, train_size, validation_size, shuffle_set_split, batch_size, plan_kwargs, **kwargs)
    247     self.trainer.fit(self._training_plan, train_dl)
    248 else:
    249     # accepts list of val dataloaders
--> 250     self.trainer.fit(self._training_plan, train_dl, val_dls)
    251 try:
    252     self.history_ = self.trainer.logger.history

File [c:\Users\resolve\.conda\envs\scvi-env\lib\site-packages\scvi\train\_trainer.py:186](file:///C:/Users/resolve/.conda/envs/scvi-env/lib/site-packages/scvi/train/_trainer.py:186), in Trainer.fit(self, *args, **kwargs)
    180 if isinstance(args[0], PyroTrainingPlan):
    181     warnings.filterwarnings(
    182         action="ignore",
    183         category=UserWarning,
    184         message="`LightningModule.configure_optimizers` returned `None`",
    185     )
--> 186 super().fit(*args, **kwargs)

File [c:\Users\resolve\.conda\envs\scvi-env\lib\site-packages\lightning\pytorch\trainer\trainer.py:529](file:///C:/Users/resolve/.conda/envs/scvi-env/lib/site-packages/lightning/pytorch/trainer/trainer.py:529), in Trainer.fit(self, model, train_dataloaders, val_dataloaders, datamodule, ckpt_path)
...
--> 318                             if self.__dict__[p].numel() == 1
    319                             else self.__dict__[p].size()) for p in param_names])
    320     return self.__class__.__name__ + '(' + args_string + ')'

AttributeError: 'NoneType' object has no attribute 'numel'

**Comment by martinkim0** [comment]
Looks like this might be an issue with Lightning. What version of Lightning do you have installed?

**Comment by simomounir** [comment]
print(lightning.__version__)
print(torch.__version__)

2.0.7
2.0.1+cu117

Is this version problematic?

Here are some specs:

**System Information** 

System: Windows
Release: 10
Version: 10.0.14393
Machine: AMD64
Processor: Intel64 Family 6 Model 85 Stepping 4, GenuineIntel

and more gpu info here:

__CUDNN VERSION: 8500
__Number CUDA Devices: 1
__CUDA Device Name: Quadro P6000
__CUDA Device Total Memory [GB]: 25.769607168

And thanks again for your help :)

**Comment by martinkim0** [comment]
Hey sorry I wasn't able to respond to this. Were you able to resolve this? If not, could you share the data you're using? Thanks

**Comment by simomounir** [comment]
No I just gave up using it because I could not find the solution to the bug.

**Comment by simomounir** [comment]
No I was not able to resolve it as I just gave up using it since I did not hear back from anyone.

I can send you instances of the objects I am using before running **GIMVI.setup_anndata()**.

**Comment by luschaef** [comment]
Got the same error with a simple scvi model (from here https://github.com/scverse/simple-scvi/ ). So I also would be very happy to hear something back about this issue :)

**Comment by MatteoSpatuzzi** [comment]
Hi, I'm experiencing the same issue with a very simple dataset. Are there any updates?

**Comment by martinkim0** [comment]
What version of scvi-tools and PyTorch Lightning do you have installed?

**Comment by MatteoSpatuzzi** [comment]
scvi-tools                1.0.4              pyhd8ed1ab_0    conda-forge
pytorch-lightning         2.1.3              pyhd8ed1ab_0    conda-forge

**Comment by martinkim0** [comment]
@simomounir @MatteoSpatuzzi sorry for the delay, looks like this is an error on the PyTorch side. I'm submitting an issue in their repo and will link it here

**Comment by martinkim0** [comment]
Nvm, it's easier to patch a fix on our end, I'll go ahead and do that.

**Comment by MatteoSpatuzzi** [comment]
Thank you!

**Comment by luschaef** [comment]
Thanks!

**Comment by canergen** [comment]
@martinkim0 can you point to the PyTorch version that doesn't have this bug? Should we fix this in the requirements for now? I guess it can take a while until it's merged into PyTorch.

**Comment by martinkim0** [comment]
It's been in the PyTorch [codebase](https://github.com/pytorch/pytorch/blob/main/torch/distributions/distribution.py#L330-L335) for a while (a couple of years). It just hasn't popped up on our end since the following need to occur:

- Model uses our implementations of negative binomial or zinb distributions
- Dataset has invalid values (non-integer and/or negative)
- Model does not pass in the `scale` argument to the distribution constructor

I've put in a pull request for PyTorch but it might just be easier and quicker to patch it on our end.

**Comment by MatteoSpatuzzi** [comment]
Hello, are there any updates on this?

**Comment by martinkim0** [comment]
It's fixed and will be available in the next release (1.1.0)
